modules/cuckooml/cuckooml.py

```diff
@@ -0,0 +1,1677 @@
+                                      simp_count], axis=1)
+
+        if not figures:
+            ret = {}
+        for f in count_dataset:
+            # Produce boxplots
+            if figures:
+                sns.boxplot(count_dataset[f])
+                sns.swarmplot(count_dataset[f], color=".25")
+                plt.title("Abnormal behaviour detection for " + f)
+                if self.context == "notebook":
+                    plt.show()
+                else:
+                    plt.savefig("abnormal_behaviour_" + f.replace(":", "_") + \
+                                ".png")
+                    plt.close()
+
+            # Get list of local outliers
+            f_1_quartile = count_dataset[f].quantile(0.25)
+            f_3_quartile = count_dataset[f].quantile(0.75)
+            f_IQR = f_3_quartile - f_1_quartile
+            f_outliers = []
+            f_suspected_outliers = []
+            for s in count_dataset[f].index:
+                if count_dataset[f][s] > f_3_quartile+3*f_IQR or \
+                        count_dataset[f][s] < f_1_quartile-3*f_IQR:
+                    f_outliers.append(s)
+                    continue
+                if count_dataset[f][s] > f_3_quartile+1.5*f_IQR or \
+                        count_dataset[f][s] < f_1_quartile-1.5*f_IQR:
+                    f_suspected_outliers.append(s)
+            if figures:
+                print f
+                print "Outliers: ", ", ".join(f_outliers)
+                print "Suspected outliers: ", ", ".join(f_suspected_outliers)
+                print "------------------------------------------------------------"
+            else:
+                ret[f] = {"outliers": f_outliers,
+                          "suspect_outliers": f_suspected_outliers}
+        if not figures:
+            return pd.DataFrame(ret).T
+
+
+    def visualise_data(self, data=None, labels=None, learning_rate=200,
+                       fig_name="custom"):
+        """Create t-Distributed Stochastic Neighbor Embedding for features and
+        labels to help inspect the data."""
+        if data is None:
+            data = self.features
+        if labels is None:
+            labels = self.labels
+
+        tsne = TSNE(learning_rate=learning_rate)
+        tsne_fit = tsne.fit_transform(data)
+        tsne_df = pd.DataFrame(tsne_fit, index=data.index, columns=['0', '1'])
+        tsne_dfl = pd.concat([tsne_df, labels], axis=1)
+
+        sns.lmplot("0", "1", data=tsne_dfl, fit_reg=False, hue="label",
+                   scatter_kws={"marker":"D", "s":50}, legend_out=True)
+        plt.title(fig_name + " (lr:" + str(learning_rate) + ")")
+        if self.context == "notebook":
+            plt.show()
+        else:
+            plt.savefig(fig_name + "_" + str(learning_rate) + ".png",
+                        bbox_inches='tight', pad_inches=1.)
+            plt.close()
+
+
+    def save_dataset(self, filename="custom_dataset.csv", features=None, \
```

Comment:
@jbremer it is saved as CSV file into current directory. How can I save it by default to the current analysis directory?